Mirror tr-design changes and update specdev markup

aphillips opened the pull request:
- Mirror the changes in tr-design
- Change `span class="uname"` to `code class="uname"` globally


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/113.html" title="Last updated on Aug 10, 2023, 3:36 PM UTC (4bdcdb5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/113/f97acd6...aphillips:4bdcdb5.html" title="Last updated on Aug 10, 2023, 3:36 PM UTC (4bdcdb5)">Diff</a>

aphillips committed:
Mirror tr-design changes and update specdev markup

- Mirror the changes in tr-design
- Change `span class="uname"` to `code class="uname"` globally

aphillips commented:
@r12a

One observation: should we recommend what to put into the `alt` for an `img`? In our example we put "NBSP" for the no-break space, but that's kind of a special case.

Also... do we want to image-ize invisibles mentioned in specdev? ("this dog food is yummy")

---

This PR changes `span` to `code`, but very little (nothing?) in the actual text about styling templates is modified here.

---

r12a commented:
I think they can probably figure out what they want to put in the alt text.  Usually, if it's not an invisible character i put the character itself, but i'm not sure we need to recommend that.

I agree that we should imagize the invisibles in specdev.

r12a commented:
I made a comment somewhere about recommending that authors style the bdi character(s) for font-family & font-size, which i imagine applies to these changes too.

aphillips committed:
Mirror the change in tr-design (styling for specific languages)

aphillips commented:
> I agree that we should imagize the invisibles in specdev.

The list is not as short as one might like 😉 but I will pull one together. If you (@r12a) could generate the images, I'd appreciate it. I think I'll make a separate task out of that...

aphillips committed:
Fix a few quirks in character markup.

aphillips committed:
Address #114: dog food using images for invisibles

aphillips committed:
Keep in sync with tr-design.

r12a commented on index.html:
```diff
@@ -2359,7 +2359,7 @@ <h5>Character naming template</h5>
 <p>The <code translate="no" class="kw">bdi</code> element is used to ensure that example characters that are right-to-left do not interfere with the layout of the page. Do not include line breaks or a space between the closing <code translate="no" class="kw">bdi</code> and the following <code translate="no" class="kw">span</code> element; spacing and presentation is controlled by styling.</p>
 
-<p>The <code translate="no" class="kw">lang</code> attribute should be filled in appropriately to get the correct font selection for a given context. Examples in East Asian languages (such as Chinese, Japanese, or Korean) or in the Arabic script can sometimes require greater care in choosing a language tag. In some cases, it might also be necessary for you to adjust the <kbd translate="no">font-f ...
+<p>The <code translate="no" class="kw">lang</code> attribute should be filled in appropriately to get the correct font selection for a given context. Examples in East Asian languages (such as Chinese, Japanese, or Korean) or in the Arabic script can sometimes require greater care in choosing a language tag. Rarely, for certain languages, it might be necessary to adjust the style of the <code>bdi< ...
```
I'm inclined to think that it's not nessecelery rare, most non-latin usages will probably need some styling.

aphillips replied:
Most users will be dealing with ASCII and perhaps some specific characters, so this "rarely" is targeted at them (rather than at us).

r12a commented:
I took one last look at the generated version.  The proposed changes look ok.  However, i noticed 2 things:

1. there is no mustard
2. the "see also" box is normally reserved for links to other sections within the document.  External links appear either in the blocks entitled "Useful background and overviews for this section" or in the pulldowns below the mustard.

aphillips committed:
Changes as discussed in 2023-08-10 teleconference
- merge subsection into section
- move text as appropriate
- make link to editing guidelines use the purple box instead of "see
  also"
- additional edits for flow and usability resulting from the above

aphillips committed:
Fix `span`->`code` and one other minor bit of text

The mustard said "...code points in the specification", but it makes
more sense to say "a specification" or perhaps "your specification".

Span-to-code was because we changed the markup but not the text.

aphillips committed:
Fix to use "useful background"
